argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -12,29 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, List, Union, Optional, Type, TYPE_CHECKING, Any
+from typing import Dict, List, Union, TYPE_CHECKING, Any
 
 from argilla.records._io._generic import GenericIO
+from argilla._helpers import resolve_hf_datasets_type
 
 if TYPE_CHECKING:
     from argilla.records import Record
 
-
-def _resolve_hf_datasets_type() -> Optional[Type["HFDataset"]]:
-    """This function resolves the `datasets.Dataset` type safely in case the datasets package is not installed.
-
-    Returns:
-        Optional[Type]: The Dataset class definition in case the datasets package is installed. Otherwise, None.
-    """
-    try:
-        from datasets import Dataset
-
-        return Dataset
-    except ImportError:
-        return None
-
-
-HFDataset = _resolve_hf_datasets_type()
+HFDataset = resolve_hf_datasets_type()
```

Comment:
(nit) Since `resolve_hf_datasets_type` is only used in this file in the end, I feel that it would make it more readable to have the conditional import only in this module (one less module to maintain).

Also I think that doing `try: import ... except ImportError: return None` in `resolve_hf_datasets_type` hides the main `ImportError` exception. Below, there is a `dataset_class = resolve_hf_datasets_type() / if dataset_class is None: / raise ImportError(...)`. Something like this:

```py
    try:
        from datasets import Dataset
    except ImportError as e:
        raise ImportError("Hugging Face datasets is not installed. Please install it using `pip install datasets`.") from e
``` 

would provide more information to the end user in case the `datasets` import fails. At some point in `huggingface_hub` we had a similar pattern and hiding the initial import gave us an headache because the `import datasets` was not failing because `datasets` was uninstalled but because a dependency of `datasets` was not correctly installed (it was not with `datasets` specifically but that's the idea).

Reply:
Thanks @Wauplin . `resolve_hf_datasets_type`'s location in `_helpers` is an artefact from sharing the function with the `hub_integration` methods.

I think we should consider making `datasets` a core dependancy because it's use has spread. @frascuchon What do you think?

Reply:
> resolve_hf_datasets_type's location in _helpers is an artefact from sharing the function with the hub_integration methods.

I think that as per https://github.com/argilla-io/argilla/pull/5161#discussion_r1672263458, this is not the case anymore (`resolve_hf_datasets_type's` seems to be only used in `argilla/src/argilla/records/_io/_datasets.py` as before this PR)